credentials/src/main/kotlin/web5/sdk/credentials/VerifiableCredential.kt

```diff
@@ -224,14 +224,9 @@ public class VerifiableCredential(public val vcDataModel: VcDataModel) {
       // using a set for fast string comparison. DIDs can be lonnng.
       val verificationMethodIds = setOf(parsedDidUrl.didUrlString, "#${parsedDidUrl.fragment}")
       val assertionMethods = didResolutionResult.didDocument.assertionMethodVerificationMethodsDereferenced
```

Comment:
i think we can leave 227-234 as is and just change the following yeah?

```suggestion
      val assertionMethods = didResolutionResult.didDocument.assertionMethodVerificationMethodsDereferenced ?: listOf()
```

Reply:
That works too. I do prefer the functional style over a `for` loop and `break` though

Reply:
```kt
val assertionMethod = assertionMethods?.firstOrNull {
  val id = it.id.toString()
  verificationMethodIds.contains(id)
}
```

this only takes the first assertion method into consideration though yeah?

Reply:
`firstOrNull` iterates over the `Iterable` and returns the first element matching the given predicate, or null if element was not found.

Reply:
oh nice! i'm just a scrub